src/main/java/com/atlassian/jira/cloud/jenkins/util/XmlUtils.java

```diff
@@ -0,0 +1,49 @@
+package com.atlassian.jira.cloud.jenkins.util;
+
+import hudson.XmlFile;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.StringReader;
+
+/**
+ * Utility class for handling XML-related operations such as parsing XML files and extracting values
+ * from XML elements.
+ *
+ * <p>This class provides methods to parse XML files, extract specific values based on tag names,
+ * and handle boolean values stored within XML elements.
+ */
+public class XmlUtils {
```

Comment:
I think we should have a few tests for this class. Would be good to cover parsing XML files, extracting values, and handling exceptions.

Reply:
Tests added